packages/core/src/storage.ts

```diff
@@ -208,6 +208,68 @@ export type FindResult<T extends Doc> = WithLookup<T>[] & {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface TxResult {}
 
+/**
+ * @public
+ */
+export interface IndexedDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  modifiedOn: Timestamp
+  modifiedBy: Ref<Account>
+  attachedTo?: Ref<Doc>
+  attachedToClass?: Ref<Class<Doc>>
+  searchTitle?: string
+  searchObjectId?: string
+  [key: string]: any
+}
+
+/**
+ * @public
+ */
+export interface SearchQuery {
+  query: string
+  classes?: Ref<Class<Doc>>[]
+  spaces?: Ref<Space>[]
+}
+
+/**
+ * @public
+ */
+export interface SearchOptions {
+  limit?: number
+}
+
+/**
+ * @public
+ */
+export interface SearchResultDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  icon?: string
+  iconComponent?: string
+  iconProps?: { [key: string]: string }
+  objectId?: string
+  title?: string
```

Comment:
IntlString and params?

Reply:
No needed for objectId and title. I've added them in IndexedDoc and fulfil them before storing the document in elastic here 
https://github.com/hcengineering/anticrm/pull/3798/files#diff-ffc4287c479a62d55793cdcd4550255db0216f7bb73bf5ada79c9811236e25dfR152

Maybe objectId is a bad naming, but I mean human readable object ID. After this issue https://github.com/hcengineering/anticrm/issues/3913 I thought that objectId is a really nice name for this.